### PR TITLE
Pull from cr.k8ssandra.io and cr.dtsx.io

### LINKED ISSUE #1636: Pull cass-operator from cr.k8ssandra.io and cr.dtsx.io
**Is your feature request related to a problem? Please describe.**
As done in cass-operator Kustomize templates also for releases, pull from cr.k8ssandra.io and cr.dtsx.io repositories instead of docker-hub directly.

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by burmanm**
**What this PR does**:
Change the image pull registeries.

**Which issue(s) this PR fixes**:
Fixes #1636 

**Checklist**
- [ ] Changes manually tested
- [ ] Automated Tests added/updated
- [ ] Documentation added/updated
- [ ] CHANGELOG.md updated (not required for documentation PRs)
- [x] CLA Signed:  [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

**Comment by Miles-Garnsey** [comment]
@burmanm is this PR still required? What is it supposed to do and is there an issue for it?

**Comment by adejanovski** [comment]
It is still required and we need to move forward with it.

**Comment by Miles-Garnsey** [comment]
I was wondering if it was done so I could approve, but there is no ticket to describe what criteria I should look for.

**Comment by burmanm** [comment]
There's similar PR on cass-operator repo, I'd merge that first before this one. Just to align the installations

**Comment by burmanm** [comment]
And this is ready for review now.. version wise, this should be merged before #1635. If done the other way, we need to modify the version to 0.46.1 here.

File: charts/cass-operator/values.yaml [hunk 1]
```diff
@@ -79,9 +79,9 @@ tolerations: []
 # systemLogger is our Vector image, configBuilder creates valid Cassandra configuration for
 # 3.11/4.0 and DSE 6.8, while k8ssandraClient is used to build configs for 4.1
 imageConfig:
-  systemLogger: k8ssandra/system-logger:v1.18.2
-  configBuilder: datastax/cass-config-builder:1.0-ubi7
-  k8ssandraClient: k8ssandra/k8ssandra-client:v0.2.2
+  systemLogger: cr.k8ssandra.io/k8ssandra/system-logger:v1.18.2
+  configBuilder: cr.dtsx.io/datastax/cass-config-builder:1.0-ubi7
```
Comment (adejanovski) [inline charts/cass-operator/values.yaml:83]: **Issue:** we'd need to get rid of the `-ubi7` here to get arm64 compatibility back and be consistent with what we do in k8ssandra-operator.

**Review by adejanovski — CHANGES_REQUESTED** [review]
Works fine, but it's probably the right occasion to bring arm64 support out of the box for cass-operator as well?

File: charts/cass-operator/values.yaml (hunk 1 above)
Comment (bradfordcp) [reply charts/cass-operator/values.yaml:83]: I wonder if we can update the cass-config-builder to ubi8 or ubi9 with ARM support.

Comment (burmanm) [reply charts/cass-operator/values.yaml:83]: @adejanovski Removing -ubi7 would be an issue as all these images are UBI images. cass-config-builder itself is an issue, but luckily entirely ignored for 4.1 and newer.

Comment (adejanovski) [reply charts/cass-operator/values.yaml:83]: We use `1.0.7` by default in k8ssandra-operator which isn't a ubi image and has arm support.

Comment (burmanm) [reply charts/cass-operator/values.yaml:83]: Yes, but since it's not an UBI image, it's not usable for us.

Comment (adejanovski) [reply charts/cass-operator/values.yaml:83]: ok, let's resolve this then

**Review by adejanovski — APPROVED** [review]

**Pull request merged**